packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/container.tsx

```diff
@@ -0,0 +1,86 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { RouteComponentProps, useLocation } from '@reach/router';
+import { currentAccount } from '../../../lib/cache';
+import { useFinishOAuthFlowHandler } from '../../../lib/oauth/hooks';
+import { Integration, useAuthClient } from '../../../models';
+import AppLayout from '../../../components/AppLayout';
+import CardHeader from '../../../components/CardHeader';
+import ConfirmSignupCode from '.';
+import { hardNavigateToContentServer } from 'fxa-react/lib/utils';
+import LoadingSpinner from 'fxa-react/components/LoadingSpinner';
+import { LocationState } from './interfaces';
+
+const SignupConfirmCodeContainer = ({
+  integration,
+}: {
+  integration: Integration;
+} & RouteComponentProps) => {
+  const authClient = useAuthClient();
+  const location = useLocation() as ReturnType<typeof useLocation> & {
+    state: LocationState;
+  };
+  const {
+    selectedNewsletterSlugs: newsletterSlugs,
+    keyFetchToken,
+    unwrapBKey,
+  } = location.state;
+
+  const { finishOAuthFlowHandler, oAuthDataError } = useFinishOAuthFlowHandler(
+    authClient,
+    integration
+  );
+
+  oAuthDataError && console.log('oAuthDataError!', oAuthDataError);
+  // TODO: UX for this, FXA-8106
+  if (oAuthDataError) {
+    return (
+      <AppLayout>
+        <CardHeader
+          headingText="Unexpected error"
+          headingTextFtlId="auth-error-999"
+        />
+      </AppLayout>
+    );
+  }
+
+  // Users who reach this page should have an account set in localStorage.
+  // Account data is added to local storage after creating an (unverified) account and after sign in.
+  // Users may also have localStorage set by the browser if they are logged in.
+
+  // The session token from local storage is required to verify the signup code.
+  // If this page is reached without an account in localStorage (e.g., directly from URL), redirect to `/`.
+
+  // TOOD: when we pull the account.verifySession call into the container component,
+  // ensure we're only reading from localStorage once. `sessionToken()` also reads from localStorage.
+
+  const storedLocalAccount = currentAccount();
+  const { email, sessionToken, uid } = storedLocalAccount;
+
+  if (!storedLocalAccount || !email || !sessionToken) {
+    hardNavigateToContentServer(`/${location.search}`);
```

Comment:
I think you may want to do something similar here [to what I'm doing on these lines](https://github.com/mozilla/fxa/pull/15858/files#diff-cafdc18fdcac681ef037d3d27b1f2b6ca094555660aae29a94379110a3d66cd1R280-R286), e.g. passing everything back to `/` except for `email` and `emailFromContent`.